float32 a_x
float32 a_y
float32 a_z

float32 w_x
float32 w_y
float32 w_z

float32 an_x
float32 an_y
float32 an_z
